scripts/nbconverted/latent_space_comparison.py

```diff
@@ -0,0 +1,320 @@
+        node_df = (genes[genes > genes.std() * high_std])
+        abbrev = 'pos'
+    elif direction == 'negative':
+        node_df = (genes[genes < -1 * (genes.std() * high_std)])
+        abbrev = 'neg'
+
+    node_df = pd.DataFrame(node_df).reset_index()
+    node_df.columns = ['genes', 'weight']
+    
+    if output_file:
+        node_df.to_csv(output_file, index=False, sep='\t')
+    
+    return node_df
+
+
+# In[17]:
+
+# Load feature matrices
+pca_feature_file = '../pancan_viz/data/pca_feature_rnaseq.tsv.gz'
+ica_feature_file = '../pancan_viz/data/ica_feature_rnaseq.tsv.gz'
+nmf_feature_file = '../pancan_viz/data/nmf_feature_rnaseq.tsv.gz'
+adage_feature_file = 'https://github.com/greenelab/tybalt/raw/928804ffd3bb3f9d5559796b2221500c303ed92c/data/adage_gene_weights.tsv'
+tybalt_feature_file = 'https://github.com/greenelab/tybalt/raw/928804ffd3bb3f9d5559796b2221500c303ed92c/results/tybalt_gene_weights.tsv'
+vae_feature_twolayer_file = 'https://github.com/greenelab/tybalt/raw/7d2854172b57efc4b92ca80d3ec86dfbbc3e4325/data/tybalt_gene_weights_twohidden100.tsv'
+vae_feature_twolayer300_file = 'https://github.com/greenelab/tybalt/raw/7d2854172b57efc4b92ca80d3ec86dfbbc3e4325/data/tybalt_gene_weights_twohidden300.tsv'
+
+pca_feature_df = pd.read_table(pca_feature_file, index_col=0)
+ica_feature_df = pd.read_table(ica_feature_file, index_col=0)
+nmf_feature_df = pd.read_table(nmf_feature_file, index_col=0)
+adage_feature_df = pd.read_table(adage_feature_file, index_col=0).T
+adage_feature_df.index = range(1, 101)
+tybalt_feature_df = pd.read_table(tybalt_feature_file, index_col=0)
+vae_tl_feature_df = pd.read_table(vae_feature_twolayer_file, index_col=0)
+vae_tl300_feature_df = pd.read_table(vae_feature_twolayer300_file, index_col=0)
+
+
+# In[18]:
+
+# This is the largest difference in the Mesenchymal subtype across various algorithms
+latent_space_df.head(1)
+
+
+# In[19]:
+
+# Define output files
+base_dir = os.path.join('results', 'feature_comparison')
+pca_out_genes = os.path.join(base_dir, 'pca_mesenchymal_genes.tsv')
+ica_out_genes = os.path.join(base_dir, 'ica_mesenchymal_genes.tsv')
+nmf_out_genes = os.path.join(base_dir, 'nmf_mesenchymal_genes.tsv')
+adage_out_genes = os.path.join(base_dir, 'adage_mesenchymal_genes.tsv')
+tybalt_out_genes = os.path.join(base_dir, 'tybalt_mesenchymal_genes.tsv')
+vae_100_out_genes = os.path.join(base_dir, 'vae100_mesenchymal_genes.tsv')
+vae_300_out_genes = os.path.join(base_dir, 'vae300_mesenchymal_genes.tsv')
+
+# Extract the most explanatory feature based on subtraction
+pca_node = latent_space_df.loc[0, 'pca_features']
+ica_node = latent_space_df.loc[0, 'ica_features']
+nmf_node = latent_space_df.loc[0, 'nmf_features']
+adage_node = latent_space_df.loc[0, 'adage_features']
+tybalt_node = latent_space_df.loc[0, 'tybalt_features']
+vae_100_node = latent_space_df.loc[0, 'vae_100_features']
+vae_300_node = latent_space_df.loc[0, 'vae_300_features']
+
+# Output the high weight genes
```

Comment:
You only want positive genes -- is that correct? I think a comment about that might be warranted

Reply:
Are the weights generally normally distributed for _all_ of these methods?

Reply:
That is a great question... I was not sure so I plotted their distributions for the respective node in the latest commit. They do not all appear to have normal distributions (particularly NMF), but I think finding high weight genes especially by standard deviations is permissible in this setting. Definitely something to strongly consider in the future though...

Reply:
👍 Yea, I think that's fine, but I thought it was definitely worth looking into.